### DISCUSSION TITLE
Import data from Atlas (MongoDB) with mongodump failed

### DISCUSSION THREAD

**Discussion by szantos** [NightscoutFoundation/xDrip#2782] [Q&A] [answered]
After processing through:
https://navid200.github.io/xDrip/docs/Nightscout/HerokuGone.html

I get following error on mongodump:
Failed: error getting collections for database `XXX_DB_USER`: error running `listCollections`. Database: `XXX_DB_USER` Err: Unsupported OP_QUERY command: listCollections. The client driver may require an upgrade. For more details see https://dochub.mongodb.org/core/legacy-opcode-removal

Can you help me here, please?

Can you confirm, is this step it the guide correct?
Do I really have to put a "user name" in place of <DATABASE>, and not the ame of the database?
2- Replace <DATABASE>, including brackets, with the user name you copied from the Atlas database access page.

**Comment by Navid200** [comment]
I will check this by running it again myself in a few hours and respond.

**Comment by Navid200** [comment]
It looks like we need to update the MongoDB release on our platform.
Thanks for reporting this.

I will be working on this and the next development branch release should address tihs.

**Comment by Navid200** [comment]
Have you updated the version of Mongodb on Atlas?
If yes, to what version?
Was that a requirement?

**Reply by szantos** [reply]
I did not perform an update.
It's 6.0.5 now.
I'm not sure what version it was before, and when/why the update happened.

**Reply by Navid200** [reply]
@tzachi-dar 
I just logged in to Atlas and can see that my MongoDB is also at 6.0.05.

But, mine is not running.  I don't think I can perform a mongodump on it to test, can I?

**Comment by jonprogers** [comment]
Hi,

I don't know if this is any help or not but I can perform a mongodump backup from atlas now that they have upgraded me to v6.0.5 as part of their upgrade program.

I do this with mongo command line tools v100.7.0 installed locally on my laptop (windows 10) from https://fastdl.mongodb.org/tools/db/mongodb-database-tools-windows-x86_64-100.7.0.zip

I then use the following command to run the dump from windows powershell:
mongodump --forceTableScan --uri <usual atlas uri starting mongodb+srv://.... and ending with the database name>

This puts the data in the dump subdirectory on my local machine.

I'm no expert, just hope communicating version numbers that do appear to work can help you make a step forward...

Good luck!!

**Reply by Navid200** [reply]
Thanks

**Comment by Navid200** [comment]
I just attempted and got the exact same error message.
Let me see what we can do to resolve this.

**Reply by goranpavicevic** [reply]
I have same problem. 6days ago worked fine and now I get same error msg. I did all packages update to newest versions, maybe that could be one of the reason and downgrading could solve the problem..?

App is running on Ubuntu 20.04 with Shared MongoDB cloud cluster. This is my mongo's versions dependencies list when grep just by mongo
_apt list --installed | grep mongo_
<img width="834" alt="image" src="https://user-images.githubusercontent.com/39703354/233321946-4ceacacf-b04e-4aa5-9103-1b67e4e37f48.png">

**Reply by jonprogers** [reply]
@goranpavicevic I first noticed that my atlas cluster had gone through the 'auto' upgrade to v6 on 13th April so that sort of ties in with your timelines assuming it took them a few days to upgrade everyone.
I don't get the error running the v100.7.0 mongodb-database-tools as posted above.

**Comment by Navid200** [comment]
It is on my list to update mongo on our platform.
We will then need to test and if everything works, we can update the dev branch.

Would you please give me a week or two?

**Comment by Navid200** [comment]
I have not forgotten about this.
I think we can update out platform to a newer release of MongoDB.   But, it is not a very simple transition.  
It will be very easy for new installation to directly use the newer version.
But, those who have an existing setup, need to have their existing database updated.
I am planning this.

But, I just want to make sure that you guys are aware of this:
https://navid200.github.io/xDrip/docs/Nightscout/NS_Transfer.html
This means that if you can have your Atlas MongoDB set up on another Nightscout host, you can transfer everything through that Nightscout site.

**Comment by Navid200** [comment, marked as answer]
You can still do this if you want to.  
The problem is that the guide that I used to have, which you can still see [here](https://navid200.github.io/xDrip/docs/Nightscout/HerokuGone.html) was for executing mongodump in Linux.  We cannot do that using the virtual machine because of the difference in mongoDB versions.
But, you can run mongodump in Windows.  This is where you can download mongodump.exe from:  https://www.mongodb.com/docs/database-tools/installation/installation-windows/

Now, you can follow the guide except that to run mongodump, you will need to use Windows.
You need to open a Windows Terminal and find the mongodump.exe that you downloaded.  Instead of typing mongodump, you need to type ./mongodump.exe if you are in the folder containing mongodump.exe.
The rest is exactly the same as before to complete mongodump.

Now, the dump folder will be on your Windows computer.  You will need to upload it into your virtual machine.
After you do that, you will be able to continue the rest of the guide in order to perform mongorestore.

**Reply by szantos** [reply]
Thanks for the detailed guide!

**Reply by Navid200** [reply]
You're welcome.
Just for anyone else who may find this through search, the guide is now updated explaining the detail for how to import Atlas MongoDB.
https://navid200.github.io/xDrip/docs/Nightscout/HerokuGone.html
